source/simulator/helper_functions.cc

```diff
@@ -1045,9 +1046,12 @@ namespace aspect
                                                                 introspection.component_indices.compositional_fields[advection_field.compositional_variable]
                                                                ),
                                                                /*dof index within component=*/ j);
-                    const double solution_value = distributed_solution(local_dof_indices[support_point_index]);
-                    const double limited_solution_value = theta * (solution_value-local_solution_average) + local_solution_average;
-                    distributed_solution(local_dof_indices[support_point_index]) = limited_solution_value;
+                    if (system_index_set.is_element(support_point_index))
```

Comment:
I can understand that the dofs at the current cell face maybe stored in the neighbor cell, therefore it may not on the current processor. However, for the DG element, if the dofs at the current cell face, then it must be store in the current cell? Perhaps that's why I didn't have any problem before?